src/flint/fmpz.jl

```diff
@@ -98,6 +99,41 @@ in(x::IntegerUnion, r::ZZRingElemUnitRange) = first(r) <= x <= last(r)
 
 mod(i::IntegerUnion, r::ZZRingElemUnitRange) = mod(i - first(r), length(r)) + first(r)
 
+struct ZZOneTo <: AbstractUnitRange{ZZRingElem}
+  stop::ZZRingElem
+end
+
+function ZZOneTo(n::ZZRingElem)
+  n <= 0 && return new(ZZ(0))
+  return new(n)
+end
+
+Base.OneTo(n::ZZRingElem) = ZZOneTo(n)
+
+Base.eltype(::Type{ZZOneTo}) = ZZRingElem
+Base.first(::ZZOneTo) = ZZ(1)
+Base.last(r::ZZOneTo) = r.stop
+Base.length(r::ZZOneTo) = Int(r.stop)
```

Comment:
```suggestion
Base.length(r::ZZOneTo) = BigInt(r.stop)
```
to match `Base.length(r::StepRange{ZZRingElem})`